### PR TITLE
Add Zalando Postgres Operator

### PR DESCRIPTION
**Pull request by joschi36**
I have tried to implement the Zalando Postgres Operator to use the library for our projects.

@Duologic Can you review the PR, I know It's maybe early as you just have merged #31. By the way, it was a complete coincidence that I started right after your merged PR.

Also, I have some troubles as I need to specify multiple CRD files. The README.md does loop over the same three `spec`, but the `output` is always the same. See here: https://github.com/jsonnet-libs/k8s/pull/36/files#diff-041ec00df6204d6beeaf7a47c00fa765b81365ec7f97aad66c5a83c95a6648d6R10-R12
Maybe there are even more problems with my current implementation.

### CODE REVIEW

**Comment by Duologic** [comment]
Cool, thanks for taking an interest.

I'm working on a more streamlined process in https://github.com/jsonnet-libs/k8s/pull/33

It still allows just one crd per spec and I personally like the simplicity of this. Would it be too crazy to merge these CRDs into a single yaml file upstream? Or break them into multiple libraries like the crossplane lib?

File: libs/zalando-postgres/config.jsonnet [hunk 1]
```diff
@@ -0,0 +1,43 @@
+local name = 'zalando-postgres';
```
Comment (Duologic) [inline libs/zalando-postgres/config.jsonnet:1]: I'm going to be arrogant here and say you can claim the 'postgress-operator' name.

I came across the https://github.com/CrunchyData/postgres-operator but that has started ~3 months later. First come, first serve? :innocent:

**Comment by Duologic** [comment]
> It still allows just one crd per spec and I personally like the simplicity of this. Would it be too crazy to merge these CRDs into a single yaml file upstream? Or break them into multiple libraries like the crossplane lib?

I had another look at this and covered this in #33. The code can now handle a `crds: []` array.

File: libs/zalando-postgres/config.jsonnet (hunk 1 above)
Comment (joschi36) [reply libs/zalando-postgres/config.jsonnet:1]: I would prefer to maybe call it `zalando-postgres-operator`. What do you think?

**Comment by joschi36** [comment]
@Duologic I have updated the PR to use the crds array you introduced in #33

File: libs/zalando-postgres/config.jsonnet (hunk 1 above)
Comment (Duologic) [reply libs/zalando-postgres/config.jsonnet:1]: Yeah, sounds fine. I'm even thinking we should drop the `-operator` suffix as this library won't install the operator.

**Comment by Duologic** [comment]
The generated files should not be committed, the config.jsonnet should be sufficient.

Heads up for #39. 😉

**Comment by xvzf** [comment]
Ping @joschi36  ;-)

**Comment by xvzf** [comment]
Closing this, please re-open it when you'd like to continue @joschi36 !

**Pull request closed without merging**